pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -1149,7 +1169,7 @@ func (b *MasqueradeBindMechanism) createNatRulesUsingNftables(proto iptables.Pro
 		}
 
 		err = Handler.NftablesAppendRule(proto, "nat", "output",
```

Comment:
Why this new rule is needed?

Reply:
when the source is 127.0.0.6 what is the dst address?

Reply:
When src IP address is 127.0.0.6, then destination IP address is IP of the virt-launcher pod

Reply:
Since I don't know what the virt-launcher pod IP is going to be (at least I think so) at this point in time, this rule is handling the `dnat` based on the source IP address.

Reply:
Maybe we could consolidate the two rules to only one rule, which would not match the destination/source address, and only the destination port?

Reply:
I don't like it since it becomes very specific to istio/envoy.
You can merge or keep it two rules but I would change it to check the port (as you mentioned) and the dst as well - we don't want to catch external destinations.
So I would change the original rule to daddr getLoopbackAdrress(proto) or podIp.

Reply:
Updated accordingly to what we agreed on Thursday - keeping the istio annotation check.
> we don't want to catch external destinations.

I have changed the rules to avoid matching external destinations:

in output chain I match the following to `dnat`:
 - destinaton address is loopback
 - destination address is pod IP
 
and in postrouting chain I match only the VM address `10.0.2.2` to `snat` to `10.0.2.1`. I think this is a better solution, it safed us one rule in postrouting and also avoids matching external destinations. Also, port-forwarding should still work as port-forwarded traffic has src and dst IP loopback address

Reply:
My comment was just about the output chain, so sounds great.